Site/webapp/wdkCustomization/js/client/components/Srt.tsx

```diff
@@ -84,37 +99,83 @@ const SUPPORTED_RECORD_CLASS_CONFIGS: InitialSrtFormConfig[] = [
     recordClassUrlSegment: 'est',
     display: 'EST IDs',
     ReporterForm: FastaGenomicSequenceReporterForm,
-    initialReporterFormState: {
+    defaultReporterFormState: {
       ...FastaGenomicSequenceReporterForm.getInitialState().formState,
       revComp: false,
       end: 200
     },
     makeInitialIdsState: () => '',
+    storedIdsState: undefined,
+    storedFormState: undefined,
     formActionUrl: '/cgi-bin/estSrt'
   },
   {
     recordClassUrlSegment: 'popsetSequence',
     display: 'Popset Isolate IDs',
     ReporterForm: FastaGenomicSequenceReporterForm,
-    initialReporterFormState: {
+    defaultReporterFormState: {
       ...FastaGenomicSequenceReporterForm.getInitialState().formState,
       revComp: false,
       end: 200
     },
     makeInitialIdsState: () => '',
+    storedIdsState: undefined,
+    storedFormState: undefined,
     formActionUrl: '/cgi-bin/isolateSrt'
   }
 ];
 
 export function Srt() {
+  const routeBase = useRouteMatch();
+  const { pathname } = useLocation();
   const compatibleSrtConfigs = useCompatibleSrtFormConfigs();
-  const [ selectedSrtForm, setSelectedSrtForm ] = useState<string>('');
+  const [selectedSrtForm, setSelectedSrtForm] = useState<string>('');
+  const [srtConfigsState, setSrtConfigsState] = useState<any>();
+  const { paramValueStore } = useNonNullableContext(WdkDependenciesContext);
+
+  useEffect(() => {
+    if (!selectedSrtForm || !srtConfigsState) return;
+    paramValueStore.updateParamValues(`srt/${selectedSrtForm}`,
+      {
+        initialIdsState: srtConfigsState[selectedSrtForm].initialIdsState,
+        initialReporterFormState: JSON.stringify(srtConfigsState[selectedSrtForm].initialReporterFormState),
+      });
+  }, [srtConfigsState]);
+
+  const updateSrtConfigsState = useCallback(
+    (idsState, formState) => {
+      const newConfigsState = {
+        ...srtConfigsState,
+        [selectedSrtForm]: {
+          initialIdsState: idsState,
+          initialReporterFormState: formState,
+        }
+      };
+      setSrtConfigsState(newConfigsState);
```

Comment:
Since you're defining the new config/tab state in terms of the current value of the state, I would use the [functional form](https://reactjs.org/docs/hooks-reference.html#functional-updates) of the `setSrtConfigsState` / `setSrtTabsState`